crates/workspace/src/workspace.rs

```diff
@@ -1927,6 +1929,46 @@ impl Workspace {
             .collect()
     }
 
+    pub fn clear_recent_files_history(
+        &mut self,
+        window: &mut Window,
+        cx: &mut Context<Workspace>,
+    ) -> Task<()> {
+        let panes = self.panes.clone();
+
+        cx.spawn_in(window, async move |workspace, cx| {
+            let answer = workspace
+                .update_in(cx, |_, window, cx| {
+                    window.prompt(
+                        PromptLevel::Warning,
+                        "Clear Recent Files History",
```

Comment:
This seems excessive: 

* Zed barely has any other actions that invoke such prompts
* A user had already did `cmd/ctrl-p` and typed a few letters to match this command; or pressed a predefined custom keybinding

Also, removing the related code will allow us to make this function synchronous, which is also good.